src/app/api/senior/route.ts

```diff
@@ -0,0 +1,111 @@
+import { withSessionAndRole } from "@server/decorator";
+import { NextResponse } from "next/server";
+import { seniorPostResponse, postSeniorSchema } from "./route.schema";
+import { prisma } from "@server/db/client";
+import { randomUUID } from "crypto";
+import { google } from "googleapis";
+import { env } from "process";
+
+export const POST = withSessionAndRole(
+  ["CHAPTER_LEADER"],
+  async ({ req, session }) => {
+    const body = await req.json();
+    const newSenior = postSeniorSchema.safeParse(body);
+
+    if (!newSenior.success) {
+      return NextResponse.json(
+        seniorPostResponse.parse({
+          code: "UNKNOWN",
+          message: "Invalid senior template",
+        }),
+        { status: 500 }
```

Comment:
If the schema fails to parse, I think it should be `INVALID_REQUEST` with status 400 (client error)